addons/OPT/GPS/fn_IsUnitLeader.sqf

```diff
@@ -28,4 +28,12 @@
 
 params ["_unit"];
 
-_unit == leader group _unit;
\ No newline at end of file
+switch (typeOf _unit) do {
+	case "OPT_NATO_Gruppenfuehrer";
+	case "OPT_CSAT_Gruppenfuehrer";
+	case "OPT_NATO_Offizier";
```

Comment:
Hast du getestet, dass dieses Fall-through funktioniert? Hab das in SQF nämlich noch nie gesehen :thinking:

Reply:
Überraschend, was SQF manchmal kann, oder? Jep, getestet und läuft :) . Verwende selten switch/case, aber manchmal kann es echt praktisch sein, zumindest wenn man solchen Krückencode überhaupt reinbringen muss.

Reply:
> Überraschend, was SQF manchmal kann, oder? 

In der Tat ^^

An dieser Stelle könnte man theoretisch auch `typeof _unit in ["OPT_NATO_Gruppenfuehrer", "OPT_CSAT_Gruppenfuehrer", ...]` verwenden. Wäre aus meiner Sicht zumindest etwas leserlicher, da es besser zeigt, was beabsichtigt wird.
Man müsste aber mal nen kleinen Performancetest der beiden Varianten machen. Hab nämlich keine Ahnung wie die da gegeneinander stehen :)

Reply:
Also ich hab das grad getestet und in meinem Test hat die `switch`-Variante etwa die dreifache Zeit der `in`-Variante benötigt (der zu findenden Eintrag war jeweils der letzte, sofern SQF die cases bzw. die Array-Einträge von vorne nach hinten durch geht):

```
Result:
0.0014 ms

Cycles:
10000/10000

Code:
test="bla" in ["some", "test", "in", "here", "and", "bla"]
```
```
Result:
0.0033 ms

Cycles:
10000/10000

Code:
test=switch("bla") do {case "some"; case "test"; case "in"; case "here"; case "and"; case "bla" : {true}; default {false}}
```